Source/WarpX.cpp

```diff
@@ -529,6 +529,28 @@ WarpX::ReadParameters ()
         pp.query("config", insitu_config);
         pp.query("pin_mesh", insitu_pin_mesh);
     }
+
+    // for slice generation //
+    {
+       ParmParse pp("slice");
+       amrex::Vector<Real> slice_lo(AMREX_SPACEDIM);
+       amrex::Vector<Real> slice_hi(AMREX_SPACEDIM);
+       Vector<int> slice_crse_ratio(AMREX_SPACEDIM);
+       pp.queryarr("dom_lo",slice_lo,0,AMREX_SPACEDIM);
+       pp.queryarr("dom_hi",slice_hi,0,AMREX_SPACEDIM);
+       pp.queryarr("coarsening_ratio",slice_crse_ratio,0,AMREX_SPACEDIM);
+       pp.query("plot_int",slice_plot_int);
+       slice_realbox.setLo(slice_lo);
+       slice_realbox.setHi(slice_hi);
+       IntVect slice_loc_ratio(AMREX_D_DECL(1,1,1));
+       slice_cr_ratio = slice_loc_ratio;
+       for (int idim = 0; idim < AMREX_SPACEDIM; ++idim)
+       {
+          slice_cr_ratio[idim] = slice_crse_ratio[idim];
+       }
```

Comment:
Thank you for catching this. 
I made a mistake here. 
I intended to copy
       for (int idim = 0; idim < AMREX_SPACEDIM; ++idim)
       {
          if (slice_crse_ratio[idim] > 0) {
              slice_cr_ratio[idim] = slice_crse_ratio[idim];
          }
       }
to initialize the coarsening ratio in each dimension to 1. So that even if the user includes coarsening ratio to be zero in the dimension which is reduced. 
For example 
if an x-z slice is required in a 3D domain, 
and the input coarsening ratio is 2, 0, 2,
The above code just ensures that the zero is not used for the y-dimension which is reduced in the slice for 1 cell length.

Reply:
I had tried passing slicee_cr_ratio to pp.queryarr() -- but it takes only std::vector<int> and not amrex::IntVect. So I read the input values to a Vector and then copy it to IntVect.  Please let me know if you have any suggestion for me here.

slice_cr_ratio is not really static since I modify it in some cases. For example -- if user had cr_ratio (2,2,2) but the dimension in x and y are reduced to one cell, then I modify cr_ratio to (1,1,2). But I wanted to defined the dimension and initialize the value. 
So I defined loc_ratio and used slice_cr_ratio = slice_loc_ratio to allocate and initialize.

When I tried to use the same strategy as that used for Ex_nodal_flag definition, the cr_ratio was required to be defined as static. 

Please let me know if you have a more elegant solution so that I can allocate, initialize, and read input values to slice_cr_ratio, while also being able to modify it if required for certain conditions.